lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -6975,13 +6975,23 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
           addToAdjointBuffer(bb, origArg, tmpBuf, loc);
           builder.emitDestroyAddrAndFold(loc, tmpBuf);
           builder.createDeallocStack(loc, tmpBuf);
-        }
-        else {
+        } else {
           recordTemporary(tan);
           addAdjointValue(bb, origArg, makeConcreteAdjointValue(tan), loc);
         }
       }
     }
+    // Record remaining pullback direct results as temporary values for cleanup.
+    // Needed to destroy unused pullback results for VJPs extracted from
+    // `@differentiable` function callees, when the `@differentiable` function's
+    // differentiation parameter indices are a superset of the active `apply`
+    // parameter indices.
+    while (allResultsIt != allResults.end()) {
+      auto unusedPullbackDirectResult = *allResultsIt++;
+      if (unusedPullbackDirectResult->getType().isAddress())
+        continue;
+      recordTemporary(unusedPullbackDirectResult);
```

Comment:
These unused results are not going to be used beyond this point. No need to record it as a temporary -- you can just `destroy_value` it.

Reply:
Thanks! Done in https://github.com/apple/swift/pull/28207/commits/a2af88f7f7653e735508076931fabc32bca195ac.